synapse/push/mailer.py

```diff
@@ -629,6 +629,27 @@ def format_ts_filter(value, format):
     return time.strftime(format, time.localtime(value / 1000))
 
 
+# XXX: This method and the next could likely be combined in a smarter way
+@staticmethod
+def load_jinja2_template(template_dir, template_filename, template_vars):
```

Comment:
I think it's confusing to have a `load_jinja2_templates` which just returns the templates, and `load_jinja2_template` which also calls `render`. I'd make a case for hoisting the `render` call to the caller which will save the `template_vars` param, but I guess you could rename the function instead.

Reply:
I've refactored `load_jinja2_templates` into a single function which can do the work of both.

The entire refactor is contained within https://github.com/matrix-org/synapse/pull/5940/commits/2ee17d70823d5b3c20b362df56d2f71418d3ac56.